index.php

```diff
@@ -28,260 +29,330 @@
-	$menu_cshow->addOption(_CHESS_SHOW_CHALLENGES_USER, _MD_CHESS_SHOW_CHALLENGES_USER);
-	$menu_cshow->addOption(_CHESS_SHOW_CHALLENGES_BOTH, _MD_CHESS_SHOW_CHALLENGES_BOTH);
-	$form1->addElement($menu_cshow);
-	$form1->addElement(new XoopsFormHidden('gstart', $gstart));
-	$form1->addElement(new XoopsFormHidden('gshow1', $gshow1));
-	$form1->addElement(new XoopsFormHidden('gshow2', $gshow2));
-	$form1->assign($xoopsTpl);
-
-	// security token not needed for this form
-	$form2 = new XoopsThemeForm('', 'form2', 'index.php');
-	$form2->addElement(new XoopsFormButton('', 'submit', _MD_CHESS_SUBMIT_BUTTON, 'submit'));
-	$menu_gshow1 = new XoopsFormSelect('', 'gshow1', $gshow1, 1, false);
-	$menu_gshow1->addOption(_CHESS_SHOW_GAMES_INPLAY,    _MD_CHESS_SHOW_GAMES_INPLAY);
-	$menu_gshow1->addOption(_CHESS_SHOW_GAMES_CONCLUDED, _MD_CHESS_SHOW_GAMES_CONCLUDED);
-	$menu_gshow1->addOption(_CHESS_SHOW_GAMES_BOTH,      _MD_CHESS_SHOW_GAMES_BOTH);
-	$form2->addElement($menu_gshow1);
-	$menu_gshow2 = new XoopsFormSelect('', 'gshow2', $gshow2, 1, false);
-	$menu_gshow2->addOption(_CHESS_SHOW_GAMES_RATED,   _MD_CHESS_SHOW_GAMES_RATED);
-	$menu_gshow2->addOption(_CHESS_SHOW_GAMES_UNRATED, _MD_CHESS_SHOW_GAMES_UNRATED);
-	$form2->addElement($menu_gshow2);
-	$form2->addElement(new XoopsFormHidden('cstart', $cstart));
-	$form2->addElement(new XoopsFormHidden('cshow',  $cshow));
-	$form2->assign($xoopsTpl);
-
-#*#DEBUG# - trying something unrelated to the chess module
-/***
-	$config_handler =& xoops_gethandler('config');
-	$clist = $config_handler->getConfigList(18);
-	var_dump('clist', $clist);
-***/
-}
-
-?>
+	"
+        )
+    );
+
+    $challenges = [];
+
+    while (false !== ($row = $xoopsDB->fetchArray($result))) {
+        $challenges[] = [
+            'challenge_id' => $row['challenge_id'],
+            'game_type'    => $row['game_type'],
+            'color_option' => $row['color_option'],
+            'player1_uid'  => $row['player1_uid'],
+            'player2_uid'  => $row['player2_uid'],
+            'create_date'  => $row['create_date'],
+            'is_rated'     => $row['is_rated'],
+        ];
+
+        // save user IDs that will require mapping to usernames
+
+        if ($row['player1_uid']) {
+            $userids[$row['player1_uid']] = 1;
+        }
+
+        if ($row['player2_uid']) {
+            $userids[$row['player2_uid']] = 1;
+        }
+    }
+
+    $xoopsDB->freeRecordSet($result);
+
+    $challenges_pagenav = new XoopsPageNav($num_challenges, $max_items_to_display, $cstart, 'cstart', "gstart=$gstart&amp;cshow=$cshow&amp;gshow1=$gshow1&amp;gshow2=$gshow2");
+
+    $xoopsTpl->assign('chess_challenges_pagenav', $challenges_pagenav->renderNav());
+
+    // ---------
+
+    // usernames
+
+    // ---------
+
+    // get mapping of user IDs to usernames
+
+    $memberHandler = xoops_getHandler('member');
+
+    $criteria = new Criteria('uid', '(' . implode(',', array_keys($userids)) . ')', 'IN');
```

Comment:
add namespace 'slash'
```suggestion
    $criteria = new \Criteria('uid', '(' . implode(',', array_keys($userids)) . ')', 'IN');
```